.. _genomics_workflows:

Genomics Workflows
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Genomics can be connected to clinical practice through the diagram shown below:

.. image:: /_static/science/Gen_clin_guidenlines.png

The process starts with acquiring patient genomic data. Next Generation Sequencing (NGS) data usually enter the system as BAM files (analyzed, mapped and annotated) with BAI index files, dependent on genome assembly versions. Microarray gene expression data format depends on the manufacturer (ie, Affy, Illumina) and chip versions. Most other data file formats are flat files, vcf, tab-delimited, often coded.

From these data, clinically relevant alleles (alternative forms of the same gene) and haplotypes (groups of alleles) are identified. 

.. image:: /_static/science/genetorec.png

Alleles and haplotypes can quickly become complex, and new discoveries are constantly made. Fortunately, we can make use of star alleles. Star alleles are haplotype patterns that have been defined at the gene level and, in many cases, associated with protein activity levels. Genetic variants within a haplotype can include single nucleotide polymorphisms (SNPs), Insertion/Deletions (InDels), and copy number variants (CNVs).

.. image:: /_static/science/haplotypes.png

Once we have applied the right nomenclature, we can then start to correlate the specific genomic characteristics of the individual with phenotypes (diseases, health conditions, or reactions ot medications) and combined with drugs and clinical trials to form clinical guidelines for doctors. 

.. image:: /_static/science/genephenotype.png

Guidelines are the implications of the gene-phenotype, gene-drug, or drug-drug interaction. For example, "increased risk of morphine formation following codeine adminstration, leading to higher risk of toxicity." These are then converted to recommendations, which actually tell the health care provider what to do. For example, "Avoid codeine use due to potential for toxicity."

.. image:: /_static/science/implic_recommend.png


Reads
@@@@@

Reads are genetic data generated by a DNA sequencing instrument,
including nucleotides and quality scores.  Reads may optionally
be aligned to a reference sequence.
(The data model for reads is similar to `SAM/BAM`_.)

.. toctree::
   reads


Variants
@@@@@@@@

Variants are genetic differences between an experimental sample and
a reference sequence.
(The data model for variants is similar to `VCF`_.)

.. toctree::
   variants


References
@@@@@@@@@@

References are standard genome sequences, used to provide a coordinate
system for reads and variants.

.. toctree::
   references

Sequence Annotations
@@@@@@@@@@@@@@@@@@@@

Sequence annotations describe genomic features such as genes and exons,
using terms from an established sequence ontology.

.. toctree::
   sequence_annotations


Allele Annotations
@@@@@@@@@@@@@@@@@@

Allele annotations are additional pieces of data often generated by
algorithms which help to describe, classify, and understand variants.

.. toctree::
   alleleAnnotations


RNA Quantification
@@@@@@@@@@@@@@@@@@@@@

The RNA quantifications provides a means of obtaining feature level
quantifications derived from a set of RNA reads.

.. toctree::
   rna_quantifications


Genotype to Phenotype
@@@@@@@@@@@@@@@@@@@@@

Genotype and phenotype data can be linked via evidence. This protocol
provides methods for describing phenotypes and associating them
with genomic features.

.. toctree::
   genotypephenotype

.. _SAM/BAM: https://samtools.github.io/hts-specs/SAMv1.pdf
.. _VCF: https://samtools.github.io/hts-specs/VCFv4.2.pdf
