dbGaP System API
!!!!!!!!!!!!!!!!!!!

Developed and operated by the National Library of Medicineâ€™s National Center for Biotechnology Information (NCBI), dbGaP archives and distributes data from studies that have investigated the relationship between phenotype and genotype, such as genome-wide association studies (GWAS). The database provides two levels of access: open (available to anyone with no restrictions), and controlled (requiring preauthorization). The controlled-access portion of the database provides for downloads of individual-level genotype and phenotype data that have been de-identified (i.e., no personal identifiers, such as name, etc.). 

The figure below shows a summary of ANNOVAR annotations:

.. image:: /_static/dbgap-summary.jpg


**Data Inputs**
@@@@@@@@@@@@@@@

**INPUTS TBD**

* Analysis
* Analysis ID
* Analysis Name
* Ancestor
* Attribution
* Belongs To
* Common Data Element Resource
* Common Data Element Term
* Data Access Committee
* Dataset
* Dataset ID
* Dataset Name
* Discriminator
* Disease
* Document
* Document ID
* Document Name
* Document Part
* Filter
* Genotype Platform
* Has Analysis
* Has Dataset
* Has Document
* Has PhenX Mapping
* Has Variable
* Is Root Study
* Is Top-level Study
* Molecular Data Type
* Object Type
* PhenX
* Primary Phenotype in Study
* Project
* Study
* Study Archive
* Study design
* Study has SRA components
* Study ID
* Study Name
* Variable
* Variable Description
* Variable ID
* Variable Name


**Required**

* TBD

**Data Outputs**
@@@@@@@@@@@@@@@@

**TBD OUTPUTS**

STUDIES

* Study - Example: phs001243.v1.p1 Genetic Basis of Breast Cancer Resistance in BRCA1 Mutation Carrier
* Embargo Release - Example: Version 1: passed embargo
* Details - Example: 14 variables, 2 documents, 2 analyses, study has SRA components
* Participants - Example: 1991
* Type of Study - Example: Case-Control
* Links - Example: BioProject Links, BioSample, Gene MeSH, SRA
* Platform - Example: SureSelectXT Human All Exon 50Mb

VARIABLES

* Clinical Variable - Example: CONSENT Genetic Basis of Breast Cancer Resistance in BRCA1 Mutation Carrier
* Dataset - Example: BRCA1_Carrier_Subject Subject ID, affection status, and consent group of participants with breast cancer and involved in the \"Genetic Basis of Breast Cancer Resistance in BRCA1 Mutation Carrier\" project.
* Variable description - Example: Consent group as determined by DAC
* Variable ID - Example: phv00273332.v1.p1

DOCUMENTS

* Study Document - Example: The Pharmacogenomics of Breast Cancer Prevention
* PGRN-RIKEN NSABP Prevention Study
* Description - Example: The Pharmacogenomics of Breast Cancer Prevention

ANALYSES

* Analysis - Example: A Genome-Wide Association Study On Pharmacogenetic Optimization Of Anticoagulant Response (POAT) Genetic And Environmental Determinants Of Warfarin Response (GEDWR) Warfarin Pharmacogenetics: Pharmacogenetic Optimization of Anticoagulant Response (POAT) and Genetic and Environmental Determinants of Warfarin Response (GEDWR) Browse genome for pha003884.1
* Description - Example: The Genetic and Environmental Determinants of Warfarin Response (GEDWR) and the Pharmacogenetic Optimization of Anticoagulation Therapy (POAT) are prospective cohort study aimed at defining the influence of CYP2C9, VKORC1 and other genes. The outcome variable we analyzed is stable warfarin dose (mg/week) that was defined as the dose (with up to a 10% change) that resulted in a therapeutic INR between 1.8 and 3.2 for at least 3 consecutive clinic visits at least two weeks apart. the number of subjects were 200.


DATASETS

* Dataset - Example: BRCA1_Carrier_Sample_Attributes Genetic Basis of Breast Cancer Resistance in BRCA1 Mutation Carrier
* Description - Example: Sample ID, anlalyte type, body site where sample was obtained, disease status, and tumor status of sample obtained from participants with breast cancer and involved in the \"Genetic Basis of Breast Cancer Resistance in BRCA1Mutation Carrier\" project.


**Required**

* TBD

**Available but not used**

* TBD