dbVar System API
!!!!!!!!!!!!!!!!!!!

The Database of Genomic Structural Variation (dbVar) is a National Center for Biotechnology Information (NCBI) archival database that manages sequence variation. dbVar complements dbSNP by archiving copy number variants (CNV), insertions, deletions, inversions, and translocations (1) that are longer than 50 base pairs (bp). The database is organized around the studies that have identified these variants, and includes variations from research-based and clinical submissions. Structural variants that have asserted germline or somatic clinical significance should be submitted to Clinvar which will forward appropriate portions of the data to dbVar for accessioning.

dbVar and the European Bioinformatics Institute’s (EBI’s) Database of Genomic Variants archive (DGVa) share the same data model and exchange data regularly. Together they represent the largest and most comprehensive archive of structural variation in the world.

Structural variation can be detected with a variety of experimental methods. Most of the data dbVar receives have been generated by Next-Generation Sequencing (NGS) or microarray (either oligo or SNP array) technologies. These methods alone can detect a wide variety of variant types, but they vary in the degree of precision and certainty they can provide with respect to breakpoint location and copy number change. A complete list of structural variation detection methods and analysis types can be found in dbVar's online documentation.

dbVar’s primary tasks are to support the submission and organization of structural variations to aid researchers in the study of a wide range of biological problems . dbVar assigns stable, traceable identifiers to the structural variants found in each study, calculates locations on newer assemblies as appropriate, provides some validation of content, and integrates the structural variant data submitted to us with a wide array of NCBI tools and data.

Some videos providing more information about dbVar are available here:

https://youtu.be/qkZiHPdyKng

https://youtu.be/8RCUU-flb9s

https://youtu.be/O92k2O28zT8


**Data Inputs**
@@@@@@@@@@@@@@@

We can access the data directly (preferred) or upload a vcf or gvf file.

**Required**

* Structural variation ID
* Object type: set to "variant" - could be variant or study
* Organism: set to "human" - could be any type of organism

**Optional (not currently used)**

* Method type
* Assertion
* Validation status
* Accession [ACC, ACCN]	
* Numeric Portion of NCBI Study ID [NST, NSTD, NSTNUM, NSTID]
* Accession Version [ACCV, ACCVERS]	
* Numeric Portion of NCBI Variant Call ID [NSSV, NSSVNUM, NSSVID]
* Alignments used for remapped placement [RALN, Alignment]
* Numeric Portion of NCBI Variant Region ID [NSV, NSVNUM, NSVID]
* Allele Origin [ALOR, ORIGIN, ALLORGN]	Object Type [OT]
* All fields [ALL]	Organism [ORG, ORGN]
* Assembly Accession [ASAC, ASSMACC, ASSMACCN, ASM_ACC]	
* Phenotype [PHEN, PHNO, PHENO, PTYPE]
* Assembly Filter [ASMF]	
* Placement rank within the major assembly [RANK]
* Assembly Name [ASSM, ASSMBLY]	
* Placement Type [PTYP, PL_TYPE]
* Assembly Organism [AORG, ASM_ORGN, ASM_ORG]
* PMID [PMID, PUBMED_ID]
* Assembly Taxonomy ID [ATAX, ASM_TAXID, ASM_TAX_ID]
* Publication Date [PDAT, PUBDAT]
* Author [AUTH]
* Remap Coverage [COVG, Coverage]
* BioProject ID [GPRJ, BIOID, BIOPROJECT]	
* Remap Failure Code [FAIL]
* BioProject Name [PRNM, BIONAM, PROJECT]	
* Sample [SMPL]
* Chromosome [CH, CHRNUM, CHROM, CHROMOSOME]
* Sample Count [SC, SCOUNT, SAMCOUNT]
* Chromosome Accession [CHRA, CHRACC, CHR_ACC, CHROM_ACCESSION]	
* Study Accession [ST, STACC, STUDY_ACCESSION]
* Chromosome End [CHRE, END, CHREND, CHR_STOP]
* Study Description [STDE, STUDY_DESC]
* Chromosome Inner End [INRE, INNER_END, CHR_INNER_STOP]	
* Study Display Name [STDN, STUDY_DISP]
* Chromosome Inner Start [INRS, INNER_START]	
* Study ID [STDY, STUDY, STUDYID]
* Chromosome Outer End [OTRE, OUTER_END, CHR_OUTER_STOP]	
* Study Type [STYP, STYPE, STUDYTYPE]
* Chromosome Outer Start [OTRS, OUTER_START]	
* Subject Phenotype status [SUPH, SUPSTA, SUB_PSTAT, SUB_PHENSTAT]
* Chromosome Start [CHRS, START, CHRSTART]
* Submitter Affiliation [LAB, CENTER]
* Detection Method [DET, DETECTION]	
* Submitter MyNCBI ID [NCBI, myNCBI]
* Filter [FILTER]	
* Submitter Name [SUB, SUBM]
* Gender [SEX,GNDR]	
* Submitter Variant ID [SVAR, SUBVAR, SVARID, SUBVARID]
* Gene Full Name [GDSC, GENED, GENEFN, GENE_DESC, GENE_FULL]	
* Taxonomy ID [TXID, TAXID, TAXONOMY_ID, TAXONOMY]
* Gene Name [GENE, GENE_NAME, SYM]	UID [UID] 
* Library Abbreviation [LIB, LIBRARY, LIBNAME]
* Unplaced Contig Accession [CTG, CONTIG, CTG_ACC, CTG_ACCESSION]
* MeSH Terms [MESH, MH]	
* Validation Method [VAL, VALIDATION]
* MeSH Unique ID [MESH_ID, MHUI, MeSHUI]
* Validation Result [VSTA, VSTAT, VALSTAT, VALIDATIONSTAT, VALIDATIONSTATUS]
* Method Platform [MPLT, METHPLAT, PLATFORM]
* Validation Result Weight [VWGT, VALSTAT_WEIGHT]
* Method Submission Name [MSUB]	
* Variant Call Accession [SSV, SSVACC, SSV_ACCESSION]
* Method Type [METH, METHOD]	
* Variant Call Count [SSVC, SSVCOUNT, SUPVARCOUNT, VC_COUNT]
* Method Type Category [MCAT, METHOD_CATEGORY]
* Variant Call Type [CLTP, VCTYPE, SSVTYPE, CALLTYPE]
* Method Type Weight [MWGT, METHOD_WEIGHT]	
* Variant Clinical Assertion [CLIN, CLIN_SIG, CLINICAL_ASSERTION, ASSERT]
* MIM_id [MIM, OMIM]	
* Variant Description [DESC]
* Modification Date [MDAT, UDAT, UPDATE, UDATE, MODATE, UPDATEDATE]	
* Variant Region Accession [SV, SVACC, SV_ACCESSION]
* Multiple in assembly [MULT]	
* Variant Region Count [VC, VCOUNT, VARCOUNT]
* Numeric Portion of EBI Study ID [ESTD, ESTNUM, ESTID]
* Variant Region Type [VT, VTYPE, VARTYPE, VRTYPE, SVTYPE]
* Numeric Portion of EBI Variant Call ID [ESSV, ESSVNUM, ESSVID]	
* Variant Size [VLEN, VSIZE, VARSIZE, VARLEN, VARLENGTH]
* Numeric Portion of EBI Variant Region ID [ESV, ESVNUM, ESVID]	
* Variant Zygosity [Zygosity, ZYG]


**Data Outputs**
@@@@@@@@@@@@@@@@

**Required**

* Clinical assertion - for example, pathogenic, likely benign, benign

**Available but not used**

Variant Region Placement Infromation

* Variant Region ID
* Placement Type
* Score
* Assembly
* Assembly Unit
* Reciprocity
* Sequence ID
* Chr
* Start
* Stop

Variant Call Information

* Variant Call ID
* Type
* Method
* Analysis
* Subject Phenotype
* Clinical
* ClinVar ID

Variant Call Placement Information

* Variant Call ID
* Placement Type
* Score
* HGVS
* Assembly
* Assembly Unit
* Reciprocity
* Sequence ID
* Chr
* Start
* Stop