.. _variants:


Variants
!!!!!!!!!!!!!!!

**1. IN USE**

* Galaxy


**2. TARGETED**

* dbSNP - Single Nucleotide Polymorphism database

* UCSC genome browser - annotated genomic reference sequences. Similar to NCBI resources but usually slower with update information as new research data are submitted to repositories appear in NCBI first. Leave out for now, or use as secondary verification source

* Ensembl - annotated genomic reference sequences. Similar to NCBI resources, slower with updated data, and uses own IDs but longer string of numbers. Tendency to remove untranslated region from mRNA sequences (ie only Start to Stop codon). Also the basis for other EMBL-EBI resources.

* ANNOVAR - Gene-, region- and filter-based annotation of genomic data.


**3. UNDER INVESTIGATION**

* AceView

* NCBI Gene - Integrative resource for genes, variants and phenotypes. Annotated genes, variants, phenotypes, linked to specific external resources. Also linked to NCBI Genome

* NCBI Genome - Integrative resource for genes, variants and phenotypes. Chromosomal assemblies, sequences, maps and annotations. Less useful than NCBI Gene at this point

* Genome Reference Consortium - Curators for genome assemblies, patches, fixes and updates. Genome data sources (for mapping ie if patientsâ€™ genomic data require analysis)

* HVP - Human Variome Project. Work to consolidate existing gene/disease specific databases. Refers back to Locus Specific Mutation Databases at Leiden (repeated below)

* LOVD Leiden Open Variation Database - Nomenclature and linking variants under different names in research

* COSMIC - Catalogue of Somatic Mutations in Cancer. Resources include Cell Lines Project (104), COSMIC, Cancer Gene Census, Drug Sensitivity, Mutational Signatures, GRCh37 Cancer Archive.

* NIEHS SNPs - SNP discovery resource focused on environmental exposures, inter-individual sequence variation, disease risk. Resequencing, data available for download. New variations deposited back to dbSNPs cos have rs IDs

* UCSC Genome assembly for positional mapping

* Seattle SNPs - SNP discovery resource for inflammatory responses. Re-sequencing methodology advice & guidance to researchers, dbSNP, RefSeq/NCBI genome assembly

* 1000 Genomes - whole genome sequencing and variants & haplotype blocks, completed project. Incorporating data into International Genome Sample Resource (IGSR) - useful

* SPSmart - large-scale genomic variants browsing (visualisation). Databases 1K Genome Project, HapMap, Perlegen, CEPH, dbSNPs

* SNAP - Retrieval of SNP LD data from 1000 genomes and HapMap. may be irrelevant at this point, researchers oriented

* Database of Genomic Variants (DGV) - structural genetic variants found in healthy individuals; CNVs, Inv

* Innate Immunity Programs for Genomic Applications - genotypes and haplotypes from genes related to innate immunity. Haplotypes tools, SNPs & InDels

* JSNP database (article downloaded into shared folder) - repo of common snps in the japanese population - weblink from article error - server not found  ?ETHNOS

* dbVar - NCBI repo of CNVs - not really CNVs. structural variations (INDEL, INV, translocations & rearrangements). Use in conjunction with dbSNPs for more complete genomic mutations

* FINDbase - Pharmacogenomic markers in different populations. OMIM, PharmGKB, Allele Freq Worldwide in ETHNOS databases

* HGDP (Human Genome Diversity Project) 


**4. REJECTED**

* Hapmap - Note: deprecated, ignore. Subsumed by 1,000 genomes project - snp genotypes from 11 different populations
